packages/engine/Source/Scene/Model/PickingPipelineStage.js

```diff
@@ -38,11 +38,11 @@ PickingPipelineStage.process = function (
   const runtimeNode = renderResources.runtimeNode;
   const shaderBuilder = renderResources.shaderBuilder;
   const model = renderResources.model;
-  const instances = runtimeNode.node.instances;
+  const instances = renderResources.instanceCount;
 
   if (renderResources.hasPropertyTable) {
     processPickTexture(renderResources, primitive, instances, context);
-  } else if (defined(instances)) {
+  } else if (defined(instances) && instances > 0) {
```

Comment:
We actually have slightly different code paths for instancing from glTF and i3dm versus instancing from the API which we have started to call runtime instancing. 
I believe `runtimeNode.node.instances` supports the glTF andf i3dm case. And it looks like  `processPickTexture` is expecting `runtimeNode.node,instances`.
So I think adding
```
const instanceCount =  renderResources.instanceCount;
```
and then modifying the lower line to 
```
} else if (defined(instances) || (defined(instanceCount) && instanceCount > 0)) {
```

would cover both cases.

Reply:
Adjusted.
It would actually fail for another case.